pkg/util/util.go

```diff
@@ -1479,3 +1479,24 @@ func GetEnvWithDefault(key string, defaultval string) string {
 	}
 	return val
 }
+
+//IsValidKubeConfigPath checks if specified `KUBECONFIG` value is valid i.e it returs true if `KUBECONFIG` env is not set
+// or if it is set, it points to a valid file
+func IsValidKubeConfigPath() bool {
+	v := os.Getenv("KUBECONFIG")
+	if v == "" {
+		u, err := user.Current()
+		if err != nil {
+			klog.V(4).Infof("failed to get user from os %s", err.Error())
+			return false
+		}
+		v = fmt.Sprintf("%s%c.kube%cconfig", u.HomeDir, os.PathSeparator, os.PathSeparator)
```

Comment:
There is a good example here for getting the path of the default kubeconfig: https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration/main.go#L45

using filepath.Join and the client-go HomeDir() function

Reply:
Ok updating. it works btw

```bash
> mv ~/.kube/config ~/.kube/configback
> ./odo catalog list components -v 4
I0601 14:27:28.755578  407430 preference.go:220] The path for preference file is /home/mzee1000/.odo/preference.yaml
I0601 14:27:28.755936  407430 util.go:728] HTTPGetRequest: https://raw.githubusercontent.com/openshift/odo/master/build/VERSION
I0601 14:27:28.755987  407430 util.go:1492] using default kubeconfig path /home/mzee1000/.kube/config
I0601 14:27:28.756013  407430 util.go:1500] invalid kubeconfig path set, KUBECONFIG env was set to /home/mzee1000/.kube/config which does no exist
I0601 14:27:28.756014  407430 util.go:749] Response will be cached in /tmp/odohttpcache for 1h0m0s
I0601 14:27:28.756074  407430 preference.go:220] The path for preference file is /home/mzee1000/.odo/preference.yaml
I0601 14:27:29.457899  407430 util.go:762] Cached response used.
Odo Devfile Components:
NAME                          DESCRIPTION                                                         REGISTRY
java-maven                    Upstream Maven and OpenJDK 11                                       DefaultDevfileRegistry
java-openliberty              Java application stack using Open Liberty runtime                   DefaultDevfileRegistry
java-quarkus                  Upstream Quarkus with Java+GraalVM                                  DefaultDevfileRegistry
java-springboot               Spring Boot® using Java                                             DefaultDevfileRegistry
java-vertx                    Upstream Vert.x using Java                                          DefaultDevfileRegistry
java-wildfly                  Upstream WildFly                                                    DefaultDevfileRegistry
java-wildfly-bootable-jar     Java stack with WildFly in bootable Jar mode, OpenJDK 11 and...     DefaultDevfileRegistry
nodejs                        Stack with NodeJS 12                                                DefaultDevfileRegistry
python                        Python Stack with Python 3.7                                        DefaultDevfileRegistry
python-django                 Python3.7 with Django                                               DefaultDevfileRegistry
I0601 14:27:29.944369  407430 odo.go:75] Could not get the latest release information in time. Never mind, exiting gracefully :)

```